### PR TITLE
Outgoint intents input

### PR DESCRIPTION
**Pull request by sloaisiga**
Support input for Outgoint intents
https://www.dashbot.io/docs/google/outgoing-intents/

### CODE REVIEW

File: src/Voxa-Dashbot.ts [hunk 1]
```diff
@@ -67,7 +95,17 @@ export function register(voxaApp: VoxaApp, config: IVoxaDashbotConfig) {
 
   voxaApp.onRequestStarted(initDashbot);
   voxaApp.onRequestStarted(trackIncoming);
-  voxaApp.onBeforeReplySent(trackOutgoing);
+  voxaApp.onRequestStarted(async voxaEvent => {
+    voxaEvent.dashbotRider = new dashbotRider();
```
Comment (armonge) [inline src/Voxa-Dashbot.ts]: We should add this to the existing `voxaEvent.dashbot` object

File: src/Voxa-Dashbot.ts [hunk 2]
```diff
@@ -67,7 +77,15 @@ export function register(voxaApp: VoxaApp, config: IVoxaDashbotConfig) {
...
   voxaApp.onRequestStarted(trackIncoming);
-  voxaApp.onBeforeReplySent(trackOutgoing);
+  voxaApp.onBeforeReplySent(async (voxaEvent, reply, transition) => {
+    let input;
+
+    if (voxaEvent.dashbot) {
+      input = voxaEvent.dashbot.input;
+    }
+
+    trackOutgoing(voxaEvent, reply, transition, input);
```
Comment (armonge) [inline src/Voxa-Dashbot.ts]: there should be an `await` here

Comment (armonge) [inline src/Voxa-Dashbot.ts]:
```suggestion
    await trackOutgoing(voxaEvent, reply, transition, input);
```

**Comment by sloaisiga** [comment]
Attaching screenshots. 
Need to look in the Hide JSON. 
[outgoing.zip](https://github.com/VoxaAI/voxa-dashbot/files/4555688/outgoing.zip)

**Comment by armonge** [comment]
@sloaisiga could you please resolve those conflicts so we get this merged?

**Comment by armonge** [comment]
@sloaisiga seems like you added a yarn.lock file, this repo uses npm. Could you please just remove yarn, it's making the build crash

**Comment by sloaisiga** [comment]
Done. @armonge would you please merge?

**Pull request merged**